reconciler/healthchecks/healthchecks.go

```diff
@@ -0,0 +1,295 @@
+package healthchecks
+
+import (
+	"sync"
+	"time"
+
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/sky-uk/merlin/types"
+)
+
+// Checker checks if the real servers of a virtual service are up.
+type Checker interface {
+	// GetDown returns the downed server IPs for the associated id. Downed servers are those that are not currently
+	// passing the health check.
+	GetDownServers(id string) []string
+	// SetHealthCheck sets the health check for the given ID. If health check is nil, it is removed.
+	SetHealthCheck(id string, check *types.VirtualService_HealthCheck) error
+	// AddServer adds a server to health check.
+	AddServer(id, ip string)
+	// RemServer removes a server from health checks.
+	RemServer(id, ip string)
+	// Stop all health checks. Use at shutdown.
+	Stop()
+}
+
+type checker struct {
+	infos map[string]*checkInfo
+	sync.Mutex
+}
+
+type checkInfo struct {
+	healthCheck    *types.VirtualService_HealthCheck
+	serverStatuses map[string]*checkStatus
+	serverStopChs  map[string]chan struct{}
+}
+
+type healthState bool
+
+const (
+	serverUp   healthState = true
+	serverDown healthState = false
+)
+
+type checkStatus struct {
+	state healthState
+	count uint32
```

Comment:
comment that count is a variable of the state, either serverUp or serverDown state.

Reply:
I think changing the variable name to something more descriptive would be better than adding a comment. Exactly what name would be better escapes me at the moment, however.